internal/indexnode/task_stats.go

```diff
@@ -564,6 +579,44 @@ func statSerializeWrite(ctx context.Context, io io.BinlogIO, startID int64, writ
 	return binlogNum, statFieldLog, nil
 }
 
+func bm25SerializerWrite(ctx context.Context, io io.BinlogIO, startID int64, writer *compaction.SegmentWriter, finalRowCount int64) (int64, []*datapb.FieldBinlog, error) {
```

Comment:
in datanode/compaction/compactor_common.go there is a function named bm25SerializeWrite

Reply:
They have different way to get logID.
Compaction and index task could use same code to do serialize, but not this pr. We need optimize binlog and stats log together.